driver/node.go

```diff
@@ -172,25 +181,55 @@ func (s *nodeService) nodePublishFilesystemVolume(req *csi.NodePublishVolumeRequ
 		return nil, err
 	}
 
-	fs, err := filesystem.New(mountOption.FsType, device)
-	if err != nil {
-		return nil, err
+	var mountOptions []string
+	if req.GetReadonly() {
+		mountOptions = append(mountOptions, "ro")
 	}
-	if !fs.Exists() {
-		if err := fs.Mkfs(); err != nil {
-			return nil, status.Errorf(codes.Internal, "failed to create filesystem: volume=%s, error=%v", req.GetVolumeId(), err)
+
+	for _, f := range mountOption.MountFlags {
+		hasOption := false
+		for _, o := range mountOptions {
+			if o == f {
+				hasOption = true
+				break
+			}
+		}
```

Comment:
In the mount command, if conflicting options are specified, the option specified later becomes effective.
There is no need to remove duplicate options.